tests/test_utils/test_map.py

```diff
@@ -0,0 +1,134 @@
+            for tiff_dtype, tiff_nodata in zip(src.dtypes, src.nodatavals):
+                assert tiff_dtype == dtype and tiff_nodata == nodata
+                assert len(set(src.block_shapes)) == 1, "tiff has multiple blocksizes"
+
+
+class TestMerge:
+    @pytest.fixture(name="output_path", scope="class")
+    def output_file_fixture(_, filesystem: FS) -> str:
+        return filesystem.getsyspath("output.tif")
+
+    @pytest.fixture(name="input_paths", scope="class")
+    def input_file_fixture(_, filesystem: FS) -> List[str]:
+        for i, bbox_coords in enumerate([(9, 9, 10, 10), (10, 9, 11, 10), (9, 10, 10, 11)]):
+            data = np.full((100, 100, 2), i, dtype="int16")
+            make_geotiff(data, BBox(bbox_coords, CRS.WGS84), filesystem, name=f"input{i}")
+        return [filesystem.getsyspath(f"input{i}.tif") for i in range(3)]
+
+    def _expected_output(_, nodata: Optional[float]) -> np.ndarray:
+        nodata = nodata or 0
+        output = np.full((200, 200, 2), nodata)
+        output[100:200, :100, ...] = 0
+        output[100:200, 100:200, ...] = 1
+        output[:100, :100, ...] = 2
+        return output
+
+    @pytest.mark.parametrize(
+        "nodata, dtype",
+        [
+            (None, "float32"),
+            (0, "float32"),
+            (-3.7, "float32"),
+            (-11, "int16"),
+            (11, "uint16"),
+            (2, "uint16"),
+            (3, "uint8"),
+        ],
+    )
+    def test_merge_tiffs(self, output_path: str, input_paths: List[str], nodata: Optional[float], dtype: str):
+        merge_tiffs(input_paths, output_path, nodata=nodata, dtype=dtype, overwrite=True)
+
+        with rasterio.open(output_path) as src:
+            for tiff_dtype, tiff_nodata in zip(src.dtypes, src.nodatavals):
+                assert tiff_dtype == dtype and tiff_nodata == approx(nodata)
+            output = np.moveaxis(src.read(), 0, -1)
+            assert_array_almost_equal(output, self._expected_output(nodata))
+
+
+class TestExtractBands:
+    @pytest.fixture(name="output_path", scope="class")
+    def output_file_fixture(_, filesystem: FS) -> str:
+        return filesystem.getsyspath("output.tif")
+
+    @pytest.fixture(name="input_data", scope="class")
+    def input_data_fixture(_) -> np.ndarray:
+        layers = [np.zeros((100, 100)), np.ones((100, 100)), np.arange(100 * 100).reshape((100, 100))]
+        return np.stack(layers, axis=-1).astype(np.int32)
+
+    @pytest.fixture(name="input_path", scope="class")
+    def input_file_fixture(_, input_data: np.ndarray, filesystem: FS) -> str:
+        make_geotiff(input_data, BBox((1, 2, 3, 4), CRS.WGS84), filesystem, name="input")
+        return filesystem.getsyspath("input.tif")
+
+    @pytest.mark.parametrize("bands", [[0], [1], [2], [0, 1], [2, 1, 0]])
+    def test_extract_bands(self, output_path: str, input_path: List[str], input_data: np.ndarray, bands: List[int]):
+        extract_bands(input_path, output_path, bands, overwrite=True)
```

Comment:
What happens with an empty list do we have an exception and should we test it ?

Reply:
if the list is empty then all of the bands are transferred over. Unlikely to be passed as an input, but we probably should raise an error in that case.